lang/java/ipc/src/main/java/org/apache/avro/ipc/HttpServer.java

```diff
@@ -48,38 +50,47 @@ public HttpServer(Responder responder, String bindAddress, int port) throws IOEx
 
   /** Constructs a server to run on the named port on the specified address. */
   public HttpServer(ResponderServlet servlet, String bindAddress, int port) throws IOException {
-    this.server = new org.mortbay.jetty.Server();
-    SelectChannelConnector connector = new SelectChannelConnector();
-    connector.setLowResourceMaxIdleTime(10000);
-    connector.setAcceptQueueSize(128);
-    connector.setResolveNames(false);
-    connector.setUseDirectBuffers(false);
+    this.server = new org.eclipse.jetty.server.Server();
+    ServerConnector connector = new ServerConnector(this.server);
     if (bindAddress != null) {
       connector.setHost(bindAddress);
     }
     connector.setPort(port);
     server.addConnector(connector);
-    new Context(server, "/").addServlet(new ServletHolder(servlet), "/*");
+
+    ServletHandler handler = new ServletHandler();
+    server.setHandler(handler);
+    handler.addServletWithMapping(new ServletHolder(servlet), "/*");
   }
 
   /** Constructs a server to run with the given connector. */
-  public HttpServer(Responder responder, Connector connector) throws IOException {
```

Comment:
Is the diff wonky or was the method with the "public HttpServer(ResponderServlet servlet, Connector connector) throws IOException" signature removed? If they were removed in my opinion it would hurt backwards compatibility in this case.